core/server/master/src/main/java/alluxio/master/meta/DefaultMetaMaster.java

```diff
@@ -341,6 +341,7 @@ public void start(Boolean isPrimary) throws IOException {
       } else {
         LOG.info("Detected existing cluster ID {}", mState.getClusterID());
       }
+      mBlockMaster.setClusterId(mState.getClusterID());
```

Comment:
Instead of MetaMaster setting the clusterID on the BlockMaster, can you make BlockMaster get clusterID from the MetaMaster every time?

Reply:
The `MetaMaster` is constructed after the `BlockMaster` is constructed, and the clusterid is generated by the `MetaMaster`.
When `BlockMaster` get  clusterId, it will wait for MetaMaster to `mBlockMaster.setClusterId`